test/implementation-report/EarlFormatter.cpp

```diff
@@ -0,0 +1,116 @@
+#include <iostream>
+#include <ctime>
+#include "EarlFormatter.h"
+
+EarlFormatter::EarlFormatter()
+{
+    // insert initial namespace
+    RdfNamespace ns;
+    ns.uri = "http://www.w3.org/ns/earl#";
+    ns.prefix = "earl";
+    namespaces.insert( ns );
+}
+
+void EarlFormatter::format( std::stringstream& ss, RdfData* data, int depth )
+{
+    // check if RdfData has a subject, if not then this is the subject and
+    // needs to be encapsulated in square brackets
+
+    // get the main object
+    auto& obj = data->subject;
+    // is the object empty? If so begin an array
+    if ( obj.name == "" )
+    {
+        ss << "[ ";
+        depth++;
+    }
+    else
+    {
+        // check for a namespace
+        if ( obj.ns.prefix != "" )
+        {
+            addNamespace( obj.ns );
+            ss << obj.ns.prefix << ':' << obj.name;
+        } else {
+            // check for last element in data
+            // TODO: better way of handling "a"
```

Comment:
check for TODO comments everywhere?

Reply:
I've removed the TODO and based on the result of the conversation below we can open an issue instead if need be